File: backend/app/models/content.py
from pydantic import BaseModel
from typing import Optional, Literal, Union

class TextContent(BaseModel):
    body: str

class VideoContent(BaseModel):
    url: str
    title: str
    caption: str = ""

class AudioContent(BaseModel):
    url: str
    title: str
    duration_seconds: int = 0

class FlashCard(BaseModel):
    card_id: str
    front: str
    back: str

class FlashcardSetContent(BaseModel):
    cards: list[FlashCard]

class QuizQuestion(BaseModel):
    question_id: str
    question_text: str
    type: Literal["multiple_choice", "true_false"]
    options: Optional[list[str]] = None
    correct_answer: str
    explanation: str

class QuizContent(BaseModel):
    questions: list[QuizQuestion]

class ContentPiece(BaseModel):
    type: Literal["text", "video", "audio", "flashcard_set", "quiz"]
    content: Union[TextContent, VideoContent, AudioContent, FlashcardSetContent, QuizContent]

class Section(BaseModel):
    section_id: str
    section_title: str
    content_pieces: list[ContentPiece]

class LessonValidation(BaseModel):
    type: Literal["multiple_choice", "true_false", "mark_complete"]
    questions: Optional[list[QuizQuestion]] = None

class GeneratedLesson(BaseModel):
    lesson_id: str
    lesson_title: str
    sections: list[Section]
    validation: LessonValidation

class GeneratedModule(BaseModel):
    module_id: str
    module_title: str
    lessons: list[GeneratedLesson]

File: frontend/src/types/content.ts
export type ContentPieceType = 'text' | 'video' | 'audio' | 'flashcard_set' | 'quiz'

export interface TextContent { body: string }
export interface VideoContent { url: string; title: string; caption: string }
export interface AudioContent { url: string; title: string; duration_seconds: number }
export interface FlashCard { card_id: string; front: string; back: string }
export interface FlashcardSetContent { cards: FlashCard[] }
export type QuestionType = 'multiple_choice' | 'true_false'
export interface QuizQuestion {
  question_id: string
  question_text: string
  type: QuestionType
  options?: string[]
  correct_answer: string
  explanation: string
}
export interface QuizContent { questions: QuizQuestion[] }

export interface ContentPiece {
  type: ContentPieceType
  content: TextContent | VideoContent | AudioContent | FlashcardSetContent | QuizContent
}

export interface Section {
  section_id: string
  section_title: string
  content_pieces: ContentPiece[]
}

export interface GeneratedLesson {
  lesson_id: string
  lesson_title: string
  sections: Section[]
  validation: {
    type: 'multiple_choice' | 'true_false' | 'mark_complete'
    questions?: QuizQuestion[]
  }
}

export interface GeneratedModule {
  module_id: string
  module_title: string
  lessons: GeneratedLesson[]
}
